### ISSUE TITLE
My Results missing location columns

### ISSUE THREAD

**Issue by andrewvanbreda** [BiologicalRecordsCentre/PoMS#122]
Missing attributes for location name, output spatial ref (i.e. to translate lat/long into grid ref), and country. 
(country can be derived from location_ids field)

**Comment by andrewvanbreda** [comment]
@BirenRathod 
Hi Biren,
Could you pull this report onto live Warehouse when you get chance. Cheers.
This isn't in use yet, so safe to pull
projects/PoMS/my_poms_results_in_detail_lingual_2.xml

**Comment by BirenRathod** [comment]
@andrewvanbreda I have pulled that report now.

**Comment by andrewvanbreda** [comment]
@CCarvell @kitenetter 
Have added this fix to the test site.

https://test-ceh-poms.pantheonsite.io/cy/my-results

There is no translation though, can you let me know if it is ok for the live site as is.

Not sure what the "output spatial ref (i.e. to translate lat/long into grid ref)" note I made was about.
I think this was after meeting with Martin.
If I am take to take this literally, that the entered grid reference should be replaced with the processed one from the cache tables, then we have the problem with won't display as the sites are sensitive, so I need to take the original from the non-cache table which is what currently happens.
So have let that requirement untouched.

Claire, are you able to let me know if this is ok for live as is, or can wait for Martin to come back.

**Comment by kitenetter** [comment]
@andrewvanbreda that works fine, just two comments:
1. The main reason for asking for an "output spatial ref" was to have the lat/long coords translated into grid references. Is there a simple way of achieving that?
2. As things stand it is difficult for the user to see how to download their records, because the download link is on the far right of the grid and can only be seen if you use the scrollbar to move to the far right. Is there a solution to that? The grid currently includes some columns that I don't think we need to display:
- Country/Island
- Other habitat

Also the font seems very large to me, but I assume that is being controlled by the theme?

**Comment by andrewvanbreda** [comment]
Hi Martin @kitenetter ,
OK, those extra columns are from the spec, but am happy to remove.
As mentioned problem with the Output_sref is that it is generated by Indicia but includes privacy_precision logic. So would be empty for our private location (although actually I don't have that column on my database for some reason, but can see the code).
I will have a go at pulling out the code directly, altering it and putting it straight in the report. 
If this is complex though I will have to leave this to include for future if/when there is further support budget for the project. 
I will let you know how I get on.

**Comment by kitenetter** [comment]
Just to clarify, we do need the columns in the download, and it is desirable to show them on the web page grid, but if it helps make the grid more useable and removes the need for the scrollbar I'm happy to drop them from the grid (but not the download).

Let me know if you need me to talk to Steve about reducing the font size in the grids.

**Comment by DavidRoy** [comment]
@BirenRathod should be able to tweak the stylesheet if font sizes need changing?

**Comment by BirenRathod** [comment]
@kitenetter I have reduced a font size, so you could able to see all column without compromising column numbers or size. see if that font size works fine or not. That also work for download link. It is on test site at present.

**Comment by kitenetter** [comment]
Thanks @BirenRathod - I think that looks much better and I'd be happy for the change to be applied to the live site.

**Comment by BirenRathod** [comment]
Thanks @kitenetter. I pushed the cod on live too.

**Comment by andrewvanbreda** [comment]
@kitenetter 
Hi
I have added the necessery configuration so those columns only appear in the download. I have added this to live also ready for when report is released. Just looking at the output_sref and will let you know shortly

**Comment by andrewvanbreda** [comment]
@kitenetter 
Ok, yes I think I misread the original comment and thought it was convert the OSGB into lat/long

We can do it the way you say, but the output_sref will be blurred by the privacy_precision, so will basically show the user what the public is seeing.

**Comment by kitenetter** [comment]
@andrewvanbreda in that case can the download include both the input coordinates (full precision, mix of grid ref and lat/long) **and** the output_sref (which I assume will always be a 1 km square?). 

No need to display output_sref in the web page grid.

**Comment by andrewvanbreda** [comment]
@kitenetter 
In theory yes, but I just noticed another problem I hadn't realised. That column is an occurrence column, not for samples, so am checking again (I had assumed it was sample as that is where the spatial references sit).
Might be able to get around this if I can use distinct in some way (as all the occurrences would have matching spatial reference).
Will get back to you about if this is going to be trouble.

**Comment by andrewvanbreda** [comment]
@kitenetter 
OK, we can't really use that column as it is an occurrence one and that causes issues (includes there being no output for it for no occurrences are recorded).
I will try to grab the output_sref code and put it directly in the report modified.
I will let you know how I get on shortly with that

**Comment by andrewvanbreda** [comment]
@BirenRathod 
Hi Biren,
Could you pull that report again, cheers
projects/PoMS/my_poms_results_in_detail_lingual_2.xml

**Comment by BirenRathod** [comment]
@andrewvanbreda That report gone live now.

**Comment by andrewvanbreda** [comment]
Hi @BirenRathod,
Thanks Biren, that has worked. I am seeing something in the data I don't understand though, can you let me know this result cheers.
select output_sref from indicia.cache_occurrences_nonfunctional where id = 20322843;

**Comment by BirenRathod** [comment]
@andrewvanbreda  It is SU5886.

**Comment by andrewvanbreda** [comment]
@BirenRathod
Thanks Biren

@kitenetter 
This is deployed to test now if you want to check it matches when you want. I am noticing something slightly odd about the output_sref. I will mail about it as I need to refer to the precise spatial references

**Comment by andrewvanbreda** [comment]
@kitenetter 
Actually I double checked, the oddity is correct. Just me misunderstanding the grid ref. I double checked visually in iRecord the spatial reference appears inside the blurred one.
Just let me know if you are happy for this to go live.

**Comment by andrewvanbreda** [comment]
@kitenetter @CCarvell 
I went ahead and put this live.

I did notice when I tested some people's My Results I was getting an error which confused me.
After investigation, I noticed this was because some users had not entered a surname, so these accounts had not been linked to the database.
These people would not have been able to enter data anyway, it is likely these accounts are just spam accounts. Never the less I have changed the system to make the Last Name mandatory to fix this issue in future.
